src/controllers/user.controller.ts

```diff
@@ -225,4 +226,71 @@ export const updatePassword = async (req: Request, res: Response) => {
       message: "An error occurred while updating the password",
     });
   }
+};
+
+export const requestPasswordReset = async (req: Request, res: Response) => {
+  try {
+    const { email } = req.body;
+
+    const user = await UserService.getUserByEmail(email);
+    if (!user) {
+      return res.status(404).json({
+        status: "fail",
+        message: "User not found",
+      });
+    }
+
+    const resetToken = await generateResetToken(user);
+    const resetLink = `${process.env.FRONTEND_URL}/reset-password?token=${resetToken}`;
+
+    const subject = "Password Reset Request";
+    const text = `Please reset your password by clicking on the following link: ${resetLink}`;
+    const html = `<p>Please reset your password by clicking on the following link:</p><a href="${resetLink}">Reset Password</a>`;
+
+    await sendEmail(email, subject, text, html);
+
+    return res.status(200).json({
+      status: "success",
+      message: "Password reset email sent",
+      data: { token: resetToken }
+    });
+  } catch (error) {
+    console.error("Error requesting password reset:", error);
+    return res.status(500).json({
+      status: "error",
+      message: "An error occurred while requesting password reset",
+    });
+  }
+};
+
+export const resetPassword = async (req: Request, res: Response) => {
+  try {
+    const newPassword = req.body.newPassword;
+    const token = req.query.token as string;
+    const decoded: any = decodeToken(token);
+    const user = await UserService.getUserByid(decoded.id);
+    console.log(decoded)
+    console.log(user)
+    if (!user) {
```

Comment:
Please remove the console logs.

Reply:
> Please remove the console logs.

Thank you, the console logs have been removed